distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSingleSignOn.java

```diff
@@ -0,0 +1,261 @@
+        // Positive = array, negative = vector, 0 (none) = single variable
+        Dictionary<String, Object> clientSettings = new Hashtable<>();
+        clientSettings.put("metadata", clientMetadata);
+        Dictionary<String, Object> serverSettings = new Hashtable<>();
+        serverSettings.put("spMetadata", new String[] {serverMetadata});
+
+        // Update the client and server with the metadata
+        Configuration clientConfig = getAdminConfig()
+                .getConfiguration("org.codice.ddf.security.idp.client.IdpMetadata", null);
+        Configuration serverConfig = getAdminConfig()
+                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
+        clientConfig.update(clientSettings);
+        serverConfig.update(serverSettings);
+        assertThat(clientConfig.getProperties(), notNullValue());
+        assertThat(serverConfig.getProperties(), notNullValue());
+    }
+
+    private String getRedirectUrl(Response response) {
+        String fullUrl = "";
+
+        // We can either get a legit redirect from the header, or we can have javascript redirect us.
+        // Whenever javascript redirects us, we have to do ugly, fragile parsing of the HTML.
+        if (response.headers().hasHeaderWithName("Location")) {
+            fullUrl = response.header("Location");
+            // TODO: Figure out how to do this using xpath
+        } else if (response.getBody().asString()
+                .contains("<title>Redirect</title>")) { // Javascript redirect
+            Pattern pattern = Pattern.compile("encoded *= *\"(.*)\"", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(response.getBody().asString());
+            matcher.find();
+            fullUrl = matcher.group(1);
+        } else {
+            // TODO: this is unexpected. Log an error?
+        }
+        return fullUrl;
+    }
+
+    private String parseUrl(Response response) {
+        String url = getRedirectUrl(response);
+
+        // We want just the base url, not all of the parameters
+        if (url.contains("?")) {
+            url = url.substring(0, url.indexOf('?'));
+        }
+        return url;
+    }
+
+    private Map<String, String> parseParams(Response response) throws Exception {
+        String url = getRedirectUrl(response);
+
+        // RestAssured expects url query parameters to be in the form of a Dictionary.
+        List<NameValuePair> paramList = URLEncodedUtils.parse(new URI(url), "UTF-8");
+        HashMap<String, String> jsonParams = new HashMap<>();
+        for (NameValuePair param : paramList) {
+            jsonParams.put(param.getName(), param.getValue());
+        }
+        return jsonParams;
+    }
+
+    @Test
+    public void testBadUsernamePassword() throws Exception {
+
+        // Request search service
+        Response searchResponse = given().
+                redirects().follow(false).
+
```

Comment:
Please reformat the extra blank lines and periods should be at the front of line of next method